app/controllers/cinema_halls_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class CinemaHallsController < ApplicationController
+  def index
+    @cinema_halls = CinemaHall.all
+    render json: @cinema_halls
+  end
+
+  def show
+    cinema_hall = CinemaHall.find(params[:id])
+    render json: cinema_hall
+  end
+
+  def create
+    @cinema_hall = CinemaHall.new(cinema_hall_params)
+    if [20, 50, 100, 200].include? @cinema_hall.size
+      @cinema_hall.save
+      cinema_hall_rows_columns
+    else
+      render json: { error: 'Wrong size' }
+    end
+  end
+
+  def edit
+    @cinema_hall = CinemaHall.find(params[:id])
+    if @cinema_hall.update(cinema_hall_params)
+      cinema_hall_rows_columns
```

Comment:
What are you trying to achieve here?

Reply:
I wanted to make it so if you change any of the params it would go through cinema_halls_rows_columns in case size was changed

Reply:
And what is the purpose of going through?

Reply:
So that you cant change the cinema size and stay with same column and row number - at least that was the idea

Reply:
Ok, now it's clear to me. So I see a few problems here:

* you're not saving the changes made by `cinema_hall_rows_columns` - they're applied after save (which is a problem on its own) but not stored, so you have inconsistency anyway
* do you really want to hard code this? shouldn't be possible to have 4 rows and 5 columns as well as 5 rows and 4 columns for 20 seats? or 2 columns and 10 rows ;)
* maybe instead of keeping the size, just keep a number of rows and columns and calculate the size on the fly?
* if you really want to make such a guard to keep the proper number of columns and rows, In a `rails way` you should move that logic to the model, and execute it with [before_save](https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-before_save) callback I guess

Reply:
Thanks, will try to fix this. As to why I want this to be hard coded - in the project specifications it is said that certain size has a set amount of rows and columns, fo: "hall 20 -> 4 seats x 5 rows"

Reply:
Ah, ok, didn't know that. So it could make sense to hard code this in model.

Reply:
Would it be fine now?

Reply:
Actually, I was thinking about linked built-in callbacks and moving the logic to the model. You can read more here: https://guides.rubyonrails.org/active_record_callbacks.html So you would end up with something like this:

```ruby
class User < ApplicationRecord
  before_save :assign_cinema_hall_rows_and_columns

  private

  def assign_cinema_hall_rows_and_columns
    # logic goes here
  end
end
```